src/libs/ajax/leonardo/providers/LeoRuntimeProvider.ts

```diff
@@ -1,17 +1,36 @@
+import _ from 'lodash/fp';
 import { isGcpContext } from 'src/analysis/utils/runtime-utils';
 import { Ajax } from 'src/libs/ajax';
 import { AbortOption } from 'src/libs/ajax/data-provider-common';
-import { ListRuntimeItem, Runtime } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { ListRuntimeItem, Runtime, RuntimeError } from 'src/libs/ajax/leonardo/models/runtime-models';
 import { AzureRuntimeWrapper, GoogleRuntimeWrapper, RuntimeWrapper } from 'src/libs/ajax/leonardo/Runtimes';
 
 export type RuntimeBasics = RuntimeWrapper & Pick<Runtime, 'cloudContext'>;
 
+export type RuntimeErrorType = 'ErrorList' | 'UserScriptError';
+export interface RuntimeErrorBase {
```

Comment:
Do we need to export the base type? Based on experience with workspace types, the base type tends to get misused in places where the union type should be used (I suspect from applying Java patterns).

```suggestion
interface RuntimeErrorBase {
```